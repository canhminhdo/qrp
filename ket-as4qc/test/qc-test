#!/bin/sh

echo "--> Testing QC"
maude < qc-test.maude -no-banner -no-advise | sed -r '/^(rewrites:|Solution|states:)/d' > qc-test.out
diff qc-test.expected qc-test.out
rm qc-test.out