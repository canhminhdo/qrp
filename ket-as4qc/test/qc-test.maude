load ../qc.maude

--- inner product
red < |0>, O > == 0 .
red < |1>, O > == 0 .
red < 0 . |0>, O > == 0 .
red < 1 . |0>, O > == 0 .
red < 2 . |0>, O > == 0 .
red < |0>, |1> > == 0 .
red < 0 . |0>, |1> > == 0 .
red < 1 . |0>, |1> > == 0 .
red < |0>, |0> > == 1 .
red < 1 . |0>, |0> > == 1 .
red < |0>, 1 . |0> > == 1 .
red < 1 . |0>, 1 . |0> > == 1 .
red < |1>, |1> > == 1 .
red < 2 . |1>, |1> > == 2 .
red < 2 . |1>, 1 . |1> > == 2 .
red < 2 . |1>, 2 . |1> > == 4 .
red < |0> (x) |0> + |1> (x) |1>, |1> (x) |1> > == 1 .
red < |0> (x) |0>, |1> (x) |1> + |0> (x) |0> > == 1 .
red < |0> (x) |0>, |1> (x) |1> > == 0 .
red < |1> (x) |0>, |1> (x) |1> > == 0 .
red < |1> (x) |1>, |1> (x) |1> > == 1 .

--- zero vector
red 0 . |0> == O .
red 0 . (0 . |0>) == O .
red 0 . (1 . |0>) == O .
red 0 . (2 . |0>) == O .
red (|0> x <1|) x O == O .
red mO x |0> == O .
red mO x 0 . |0> == O .
red mO x 1 . |0> == O .
red mO x 2 . |0> == O .
red |0> (x) O == O .
red 0 . |0> (x) O == O .
red 1 . |0> (x) O == O .
red 2 . |0> (x) O == O .
red O (x) 0 . |1> == O .
red O (x) 1 . |1> == O .
red O (x) 2 . |1> == O .

--- zero covector
red 0 . <0| == O^+ .
red 0 .  0 . <0| == O^+ .
red 0 .  1 . <0| == O^+ .
red 0 .  2 . <0| == O^+ .
red 0 . O^+ == O^+ . 
red 1 . O^+ == O^+ . 
red 2 . O^+ == O^+ . 
red O^+ + 0 . <0| == 0 . <0| . 
red O^+ + 1 . <0| == 1 . <0| .
red O^+ + 2 . <0| == 2 . <0| .
red (0 . |0>) x O^+ == mO .
red (1 . |0>) x O^+ == mO .
red (2 . |0>) x O^+ == mO .
red O x (0 . <0|) == mO .
red O x (1 . <0|) == mO .
red O x (2 . <0|) == mO .
red (0 . <0|) (x) O^+ == O^+ .
red (1 . <0|) (x) O^+ == O^+ .
red (2 . <0|) (x) O^+ == O^+ .
red O^+ (x) (0 . <0|) == O^+ .
red O^+ (x) (1 . <0|) == O^+ .
red O^+ (x) (2 . <0|) == O^+ .

--- zero matrix
red 0 . |0><0| == mO .
red 0 . 0 . |0><0| == mO .
red 0 . 1 . |0><0| == mO .
red 0 . 2 . |0><0| == mO .
red 0 . mO == mO .
red 1 . mO == mO .
red 2 . mO == mO .
red |0><0| x mO == mO .
red (0 . |0><0|) x mO == mO .
red (1 . |0><0|) x mO == mO .
red (2 . |0><0|) x mO == mO .
red mO x (0 . |0><0|) == mO .
red mO x (1 . |0><0|) == mO .
red mO x (2 . |0><0|) == mO .
red mO (x) (|0><0| (x) |1><1|) == mO .
red mO (x) (0 . |0><0| (x) |1><1|) == mO .
red mO (x) (1 . |0><0| (x) |1><1|) == mO .
red mO (x) (2 . |0><0| (x) |1><1|) == mO .
red mO (x) (|0><0| (x) 0 . |1><1|) == mO .
red mO (x) (|0><0| (x) 1 . |1><1|) == mO .
red mO (x) (|0><0| (x) 2 . |1><1|) == mO .
red mO (x) (0 . |0><0| (x) 0 . |1><1|) == mO .
red mO (x) (1 . |0><0| (x) 1 . |1><1|) == mO .
red mO (x) (2 . |0><0| (x) 2 . |1><1|) == mO .
red (|0><0| (x) |1><1|) (x) mO == mO .
red (0 . |0><0| (x) |1><1|) (x) mO == mO .
red (1 . |0><0| (x) |1><1|) (x) mO == mO .
red (2 . |0><0| (x) |1><1|) (x) mO == mO .
red (|0><0| (x) 0 . |1><1|) (x) mO == mO .
red (|0><0| (x) 1 . |1><1|) (x) mO == mO .
red (|0><0| (x) 2 . |1><1|) (x) mO == mO .
red (0 . |0><0| (x) 0 . |1><1|) (x) mO == mO .
red (1 . |0><0| (x) 1 . |1><1|) (x) mO == mO .
red (2 . |0><0| (x) 2 . |1><1|) (x) mO == mO .

--- identity matrix properties
red I x |0><0| == |0><0| .
red I x (0 . |0><0|) == 0 . |0><0| .
red I x (1 . |0><0|) == 1 . |0><0| .
red I x (2 . |0><0|) == 2 . |0><0| .
red I x mO == mO .
red |0><0| x I == |0><0| .
red (0 . |0><0|) x I == 0 . |0><0| .
red (1 . |0><0|) x I == 1 . |0><0| .
red (2 . |0><0|) x I == 2 . |0><0| .
red mO x I == mO .

--- dagger properties
red (0 . |0>)^+ == O^+ .
red (1 . |0>)^+ == <0| .
red (2 . |0>)^+ == 2 . <0| .
red (0 . |0> + 0 . |1>)^+ == O^+ .
red (0 . |0> + |1>)^+ == <1| .
red (|0> + 0 . |1>)^+ == <0| .
red (|0> + |1>)^+ == <0| + <1| .
red (1 . |0> + 1 . |1>)^+ == <0| + <1| .
red (2 . |0> + 2 . |1>)^+ == 2 . <0| + 2 . <1| .
red (0 . |0> (x) |1>)^+ == O^+ .
red (|0> (x) 0 . |1>)^+ == O^+ .
red (0 . |0> (x) 0 . |1>)^+ == O^+ .
red (|0> (x) |1>)^+ == <0| (x) <1| .
red (1 . |0> (x) |1>)^+ == <0| (x) <1| .
red (|0> (x) 1 . |1>)^+ == <0| (x) <1| .
red (1 . |0> (x) 1 . |1>)^+ == <0| (x) <1| .
red (0 . <0|)^+ == O .
red (1 . <0|)^+ == |0> .
red (2 . <0|)^+ == 2 . |0> .
red (0 . <0| + 0 . <1|)^+ == O .
red (0 . <0| + <1|)^+ == |1> .
red (<0| + 0 . <1|)^+ == |0> .
red (<0| + <1|)^+ == |0> + |1> .
red (1 . <0| + 1 . <1|)^+ == |0> + |1> .
red (2 . <0| + 2 . <1|)^+ == 2 . |0> + 2 . |1> .
red (0 . <0| (x) <1|)^+ == O .
red (<0| (x) 0 . <1|)^+ == O .
red (0 . <0| (x) 0 . <1|)^+ == O .
red (<0| (x) <1|)^+ == |0> (x) |1> .
red (1 . <0| (x) <1|)^+ == |0> (x) |1> .
red (<0| (x) 1 . <1|)^+ == |0> (x) |1> .
red (1 . <0| (x) 1 . <1|)^+ == |0> (x) |1> .

--- multiplication distributes over addition
red |0><0| x (|0> + |1>) == |0> .
red |0><0| x (1 . |0> + |1>) == |0> .
red |0><0| x (2 . |0> + |1>) == 2 . |0> .
red (|0><0| + |1><1|) x |0> == |0> .
red (1 . |0><0| + |1><1|) x |0> == |0> .
red (2 . |0><0| + |1><1|) x |0> == 2 . |0> .
red (|0><0| + |1><1|) x (|0> + |1>) == |0> + |1> .
red (1 . |0><0| + |1><1|) x (|0> + |1>) == |0> + |1> .
red (|0><0| + 1 . |1><1|) x (|0> + |1>) == |0> + |1> .
red (2 . |0><0| + 2 . |1><1|) x (|0> + |1>) == 2 . |0> + 2 . |1> .
red |0> x (<0| + <1|) == |0> x <0| + |0> x <1| .
red |0> x (1 . <0| + 1 . <1|) == |0> x <0| + |0> x <1| .
red |0> x (0 . <0| + 0 . <1|) == mO .
red |0> x (0 . <0| + 1 . <1|) == |0> x <1| .
red |0> x (0 . <0| + 2 . <1|) == 2 . (|0> x <1|) .
red (|0> + |1>) x <0| == |0> x <0| + |1> x <0| .
red (1 . |0> + 1 . |1>) x <0| == |0> x <0| + |1> x <0| .
red (0 . |0> + 0 . |1>) x <0| == mO .
red (0 . |0> + 1 . |1>) x <0| == |1> x <0| .
red (0 . |0> + 2 . |1>) x <0| == 2 . (|1> x <0|) .
red (|0> + |1>) x (<0| + <1|) == |0> x <0| + |0> x <1| + |1> x <0| + |1> x <1| .
red (0 . |0> + 0 . |1>) x (<0| + <1|) == mO .
red (0 . |0> + |1>) x (<0| + <1|) == |1> x <0| + |1> x <1| .
red (1 . |0> + 1 . |1>) x (<0| + <1|) == |0> x <0| + |0> x <1| + |1> x <0| + |1> x <1| .
red (2 . |0> + 1 . |1>) x (<0| + <1|) == 2 . (|0> x <0|) + 2 . (|0> x <1|) + |1> x <0| + |1> x <1| .
red (|0> x <1|) x (|0> x <0| + |1> x <1|) == |0> x <1| .
red (|0> x <1|) x (|0> x <0| + 0 . (|1> x <1|)) == mO .
red (|0> x <1|) x (|0> x <0| + 1 . (|1> x <1|)) == 1 . (|0> x <1|) .
red (|0> x <1|) x (|0> x <0| + 2 . (|1> x <1|)) == 2 . (|0> x <1|) .
red (|0> x <0| + |1> x <1|) x (|0> x <1|) == |0> x <1| .
red (0 . (|0> x <0|) + |1> x <1|) x (|0> x <1|) == mO .
red (1 . (|0> x <0|) + |1> x <1|) x (|0> x <1|) == 1 . (|0> x <1|) .
red (2 . (|0> x <0|) + |1> x <1|) x (|0> x <1|) == 2 . (|0> x <1|) .
red (|0> x <0| + |1> x <1|) x (|0> x <0| + |1> x <1|) == |0> x <0| + |1> x <1| .

--- scalar normal forms

--- additive normal forms

--- multiplicative normal forms

--- tensor product normal forms

--- tensor distributes over addition