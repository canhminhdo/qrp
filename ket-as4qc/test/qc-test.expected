==========================================
reduce in QC : < |0>,O > == 0 .
result Bool: true
==========================================
reduce in QC : < |1>,O > == 0 .
result Bool: true
==========================================
reduce in QC : < 0 . |0>,O > == 0 .
result Bool: true
==========================================
reduce in QC : < 1 . |0>,O > == 0 .
result Bool: true
==========================================
reduce in QC : < 2 . |0>,O > == 0 .
result Bool: true
==========================================
reduce in QC : < |0>,|1> > == 0 .
result Bool: true
==========================================
reduce in QC : < 0 . |0>,|1> > == 0 .
result Bool: true
==========================================
reduce in QC : < 1 . |0>,|1> > == 0 .
result Bool: true
==========================================
reduce in QC : < |0>,|0> > == 1 .
result Bool: true
==========================================
reduce in QC : < 1 . |0>,|0> > == 1 .
result Bool: true
==========================================
reduce in QC : < |0>,1 . |0> > == 1 .
result Bool: true
==========================================
reduce in QC : < 1 . |0>,1 . |0> > == 1 .
result Bool: true
==========================================
reduce in QC : < |1>,|1> > == 1 .
result Bool: true
==========================================
reduce in QC : < 2 . |1>,|1> > == 2 .
result Bool: true
==========================================
reduce in QC : < 2 . |1>,1 . |1> > == 2 .
result Bool: true
==========================================
reduce in QC : < 2 . |1>,2 . |1> > == 4 .
result Bool: true
==========================================
reduce in QC : < |0>(x)|0> + |1>(x)|1>,|1>(x)|1> > == 1 .
result Bool: true
==========================================
reduce in QC : < |0>(x)|0>,|0>(x)|0> + |1>(x)|1> > == 1 .
result Bool: true
==========================================
reduce in QC : < |0>(x)|0>,|1>(x)|1> > == 0 .
result Bool: true
==========================================
reduce in QC : < |1>(x)|0>,|1>(x)|1> > == 0 .
result Bool: true
==========================================
reduce in QC : < |1>(x)|1>,|1>(x)|1> > == 1 .
result Bool: true
==========================================
reduce in QC : 0 . |0> == O .
result Bool: true
==========================================
reduce in QC : 0 . 0 . |0> == O .
result Bool: true
==========================================
reduce in QC : 0 . 1 . |0> == O .
result Bool: true
==========================================
reduce in QC : 0 . 2 . |0> == O .
result Bool: true
==========================================
reduce in QC : |0> x <1| x O == O .
result Bool: true
==========================================
reduce in QC : mO x |0> == O .
result Bool: true
==========================================
reduce in QC : mO x 0 . |0> == O .
result Bool: true
==========================================
reduce in QC : mO x 1 . |0> == O .
result Bool: true
==========================================
reduce in QC : mO x 2 . |0> == O .
result Bool: true
==========================================
reduce in QC : |0>(x)O == O .
result Bool: true
==========================================
reduce in QC : 0 . |0>(x)O == O .
result Bool: true
==========================================
reduce in QC : 1 . |0>(x)O == O .
result Bool: true
==========================================
reduce in QC : 2 . |0>(x)O == O .
result Bool: true
==========================================
reduce in QC : O(x)0 . |1> == O .
result Bool: true
==========================================
reduce in QC : O(x)1 . |1> == O .
result Bool: true
==========================================
reduce in QC : O(x)2 . |1> == O .
result Bool: true
==========================================
reduce in QC : 0 . <0| == O^+ .
result Bool: true
==========================================
reduce in QC : 0 . 0 . <0| == O^+ .
result Bool: true
==========================================
reduce in QC : 0 . 1 . <0| == O^+ .
result Bool: true
==========================================
reduce in QC : 0 . 2 . <0| == O^+ .
result Bool: true
==========================================
reduce in QC : 0 . O^+ == O^+ .
result Bool: true
==========================================
reduce in QC : 1 . O^+ == O^+ .
result Bool: true
==========================================
reduce in QC : 2 . O^+ == O^+ .
result Bool: true
==========================================
reduce in QC : O^+ + 0 . <0| == 0 . <0| .
result Bool: true
==========================================
reduce in QC : O^+ + 1 . <0| == 1 . <0| .
result Bool: true
==========================================
reduce in QC : O^+ + 2 . <0| == 2 . <0| .
result Bool: true
==========================================
reduce in QC : (0 . |0>) x O^+ == mO .
result Bool: true
==========================================
reduce in QC : (1 . |0>) x O^+ == mO .
result Bool: true
==========================================
reduce in QC : (2 . |0>) x O^+ == mO .
result Bool: true
==========================================
reduce in QC : O x 0 . <0| == mO .
result Bool: true
==========================================
reduce in QC : O x 1 . <0| == mO .
result Bool: true
==========================================
reduce in QC : O x 2 . <0| == mO .
result Bool: true
==========================================
reduce in QC : (0 . <0|)(x)O^+ == O^+ .
result Bool: true
==========================================
reduce in QC : (1 . <0|)(x)O^+ == O^+ .
result Bool: true
==========================================
reduce in QC : (2 . <0|)(x)O^+ == O^+ .
result Bool: true
==========================================
reduce in QC : O^+(x)0 . <0| == O^+ .
result Bool: true
==========================================
reduce in QC : O^+(x)1 . <0| == O^+ .
result Bool: true
==========================================
reduce in QC : O^+(x)2 . <0| == O^+ .
result Bool: true
==========================================
reduce in QC : 0 . |0><0| == mO .
result Bool: true
==========================================
reduce in QC : 0 . 0 . |0><0| == mO .
result Bool: true
==========================================
reduce in QC : 0 . 1 . |0><0| == mO .
result Bool: true
==========================================
reduce in QC : 0 . 2 . |0><0| == mO .
result Bool: true
==========================================
reduce in QC : 0 . mO == mO .
result Bool: true
==========================================
reduce in QC : 1 . mO == mO .
result Bool: true
==========================================
reduce in QC : 2 . mO == mO .
result Bool: true
==========================================
reduce in QC : |0><0| x mO == mO .
result Bool: true
==========================================
reduce in QC : (0 . |0><0|) x mO == mO .
result Bool: true
==========================================
reduce in QC : (1 . |0><0|) x mO == mO .
result Bool: true
==========================================
reduce in QC : (2 . |0><0|) x mO == mO .
result Bool: true
==========================================
reduce in QC : mO x 0 . |0><0| == mO .
result Bool: true
==========================================
reduce in QC : mO x 1 . |0><0| == mO .
result Bool: true
==========================================
reduce in QC : mO x 2 . |0><0| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)|0><0|(x)|1><1| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)0 . |0><0|(x)|1><1| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)1 . |0><0|(x)|1><1| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)2 . |0><0|(x)|1><1| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)|0><0|(x)0 . |1><1| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)|0><0|(x)1 . |1><1| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)|0><0|(x)2 . |1><1| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)0 . |0><0|(x)0 . |1><1| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)1 . |0><0|(x)1 . |1><1| == mO .
result Bool: true
==========================================
reduce in QC : mO(x)2 . |0><0|(x)2 . |1><1| == mO .
result Bool: true
==========================================
reduce in QC : (|0><0|(x)|1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : (0 . |0><0|(x)|1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : (1 . |0><0|(x)|1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : (2 . |0><0|(x)|1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : (|0><0|(x)0 . |1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : (|0><0|(x)1 . |1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : (|0><0|(x)2 . |1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : (0 . |0><0|(x)0 . |1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : (1 . |0><0|(x)1 . |1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : (2 . |0><0|(x)2 . |1><1|)(x)mO == mO .
result Bool: true
==========================================
reduce in QC : I x |0><0| == |0><0| .
result Bool: true
==========================================
reduce in QC : I x 0 . |0><0| == 0 . |0><0| .
result Bool: true
==========================================
reduce in QC : I x 1 . |0><0| == 1 . |0><0| .
result Bool: true
==========================================
reduce in QC : I x 2 . |0><0| == 2 . |0><0| .
result Bool: true
==========================================
reduce in QC : I x mO == mO .
result Bool: true
==========================================
reduce in QC : |0><0| x I == |0><0| .
result Bool: true
==========================================
reduce in QC : (0 . |0><0|) x I == 0 . |0><0| .
result Bool: true
==========================================
reduce in QC : (1 . |0><0|) x I == 1 . |0><0| .
result Bool: true
==========================================
reduce in QC : (2 . |0><0|) x I == 2 . |0><0| .
result Bool: true
==========================================
reduce in QC : mO x I == mO .
result Bool: true
==========================================
reduce in QC : (0 . |0>)^+ == O^+ .
result Bool: true
==========================================
reduce in QC : (1 . |0>)^+ == <0| .
result Bool: true
==========================================
reduce in QC : (2 . |0>)^+ == 2 . <0| .
result Bool: true
==========================================
reduce in QC : (0 . |0> + 0 . |1>)^+ == O^+ .
result Bool: true
==========================================
reduce in QC : (|1> + 0 . |0>)^+ == <1| .
result Bool: true
==========================================
reduce in QC : (|0> + 0 . |1>)^+ == <0| .
result Bool: true
==========================================
reduce in QC : (|0> + |1>)^+ == <0| + <1| .
result Bool: true
==========================================
reduce in QC : (1 . |0> + 1 . |1>)^+ == <0| + <1| .
result Bool: true
==========================================
reduce in QC : (2 . |0> + 2 . |1>)^+ == 2 . <0| + 2 . <1| .
result Bool: true
==========================================
reduce in QC : (0 . |0>(x)|1>)^+ == O^+ .
result Bool: true
==========================================
reduce in QC : (|0>(x)0 . |1>)^+ == O^+ .
result Bool: true
==========================================
reduce in QC : (0 . |0>(x)0 . |1>)^+ == O^+ .
result Bool: true
==========================================
reduce in QC : (|0>(x)|1>)^+ == <0|(x)<1| .
result Bool: true
==========================================
reduce in QC : (1 . |0>(x)|1>)^+ == <0|(x)<1| .
result Bool: true
==========================================
reduce in QC : (|0>(x)1 . |1>)^+ == <0|(x)<1| .
result Bool: true
==========================================
reduce in QC : (1 . |0>(x)1 . |1>)^+ == <0|(x)<1| .
result Bool: true
==========================================
reduce in QC : (0 . <0|)^+ == O .
result Bool: true
==========================================
reduce in QC : (1 . <0|)^+ == |0> .
result Bool: true
==========================================
reduce in QC : (2 . <0|)^+ == 2 . |0> .
result Bool: true
==========================================
reduce in QC : (0 . <0| + 0 . <1|)^+ == O .
result Bool: true
==========================================
reduce in QC : (<1| + 0 . <0|)^+ == |1> .
result Bool: true
==========================================
reduce in QC : (<0| + 0 . <1|)^+ == |0> .
result Bool: true
==========================================
reduce in QC : (<0| + <1|)^+ == |0> + |1> .
result Bool: true
==========================================
reduce in QC : (1 . <0| + 1 . <1|)^+ == |0> + |1> .
result Bool: true
==========================================
reduce in QC : (2 . <0| + 2 . <1|)^+ == 2 . |0> + 2 . |1> .
result Bool: true
==========================================
reduce in QC : (0 . <0|(x)<1|)^+ == O .
result Bool: true
==========================================
reduce in QC : (<0|(x)0 . <1|)^+ == O .
result Bool: true
==========================================
reduce in QC : (0 . <0|(x)0 . <1|)^+ == O .
result Bool: true
==========================================
reduce in QC : (<0|(x)<1|)^+ == |0>(x)|1> .
result Bool: true
==========================================
reduce in QC : (1 . <0|(x)<1|)^+ == |0>(x)|1> .
result Bool: true
==========================================
reduce in QC : (<0|(x)1 . <1|)^+ == |0>(x)|1> .
result Bool: true
==========================================
reduce in QC : (1 . <0|(x)1 . <1|)^+ == |0>(x)|1> .
result Bool: true
==========================================
reduce in QC : |0><0| x (|0> + |1>) == |0> .
result Bool: true
==========================================
reduce in QC : |0><0| x (|1> + 1 . |0>) == |0> .
result Bool: true
==========================================
reduce in QC : |0><0| x (|1> + 2 . |0>) == 2 . |0> .
result Bool: true
==========================================
reduce in QC : (|0><0| + |1><1|) x |0> == |0> .
result Bool: true
==========================================
reduce in QC : (|1><1| + 1 . |0><0|) x |0> == |0> .
result Bool: true
==========================================
reduce in QC : (|1><1| + 2 . |0><0|) x |0> == 2 . |0> .
result Bool: true
==========================================
reduce in QC : (|0><0| + |1><1|) x (|0> + |1>) == |0> + |1> .
result Bool: true
==========================================
reduce in QC : (|1><1| + 1 . |0><0|) x (|0> + |1>) == |0> + |1> .
result Bool: true
==========================================
reduce in QC : (|0><0| + 1 . |1><1|) x (|0> + |1>) == |0> + |1> .
result Bool: true
==========================================
reduce in QC : (2 . |0><0| + 2 . |1><1|) x (|0> + |1>) == 2 . |0> + 2 . |1> .
result Bool: true
==========================================
reduce in QC : |0> x (<0| + <1|) == |0> x <0| + |0> x <1| .
result Bool: true
==========================================
reduce in QC : |0> x (1 . <0| + 1 . <1|) == |0> x <0| + |0> x <1| .
result Bool: true
==========================================
reduce in QC : |0> x (0 . <0| + 0 . <1|) == mO .
result Bool: true
==========================================
reduce in QC : |0> x (0 . <0| + 1 . <1|) == |0> x <1| .
result Bool: true
==========================================
reduce in QC : |0> x (0 . <0| + 2 . <1|) == 2 . |0> x <1| .
result Bool: true
==========================================
reduce in QC : (|0> + |1>) x <0| == |0> x <0| + |1> x <0| .
result Bool: true
==========================================
reduce in QC : (1 . |0> + 1 . |1>) x <0| == |0> x <0| + |1> x <0| .
result Bool: true
==========================================
reduce in QC : (0 . |0> + 0 . |1>) x <0| == mO .
result Bool: true
==========================================
reduce in QC : (0 . |0> + 1 . |1>) x <0| == |1> x <0| .
result Bool: true
==========================================
reduce in QC : (0 . |0> + 2 . |1>) x <0| == 2 . |1> x <0| .
result Bool: true
==========================================
reduce in QC : (|0> + |1>) x (<0| + <1|) == ((|1> x <0| + |1> x <1|) + |0> x
    <1|) + |0> x <0| .
result Bool: true
==========================================
reduce in QC : (0 . |0> + 0 . |1>) x (<0| + <1|) == mO .
result Bool: true
==========================================
reduce in QC : (|1> + 0 . |0>) x (<0| + <1|) == |1> x <0| + |1> x <1| .
result Bool: true
==========================================
reduce in QC : (1 . |0> + 1 . |1>) x (<0| + <1|) == ((|1> x <0| + |1> x <1|) +
    |0> x <1|) + |0> x <0| .
result Bool: true
==========================================
reduce in QC : (1 . |1> + 2 . |0>) x (<0| + <1|) == 2 . |0> x <0| + 2 . |0> x
    <1| + |1> x <0| + |1> x <1| .
result Bool: true
==========================================
reduce in QC : (|0> x <1|) x (|0> x <0| + |1> x <1|) == |0> x <1| .
result Bool: true
==========================================
reduce in QC : (|0> x <1|) x (0 . |1> x <1| + |0> x <0|) == mO .
result Bool: true
==========================================
reduce in QC : (|0> x <1|) x (1 . |1> x <1| + |0> x <0|) == 1 . |0> x <1| .
result Bool: true
==========================================
reduce in QC : (|0> x <1|) x (2 . |1> x <1| + |0> x <0|) == 2 . |0> x <1| .
result Bool: true
==========================================
reduce in QC : (|0> x <0| + |1> x <1|) x |0> x <1| == |0> x <1| .
result Bool: true
==========================================
reduce in QC : (0 . |0> x <0| + |1> x <1|) x |0> x <1| == mO .
result Bool: true
==========================================
reduce in QC : (1 . |0> x <0| + |1> x <1|) x |0> x <1| == 1 . |0> x <1| .
result Bool: true
==========================================
reduce in QC : (2 . |0> x <0| + |1> x <1|) x |0> x <1| == 2 . |0> x <1| .
result Bool: true
==========================================
reduce in QC : (|0> x <0| + |1> x <1|) x (|0> x <0| + |1> x <1|) == |0> x <0| +
    |1> x <1| .
result Bool: true
Bye.
